### PR TITLE
Fixed a bug where a user without any roles wont be able to load the crewpage because mappings[0] is undefined, added a if/else for this to display "Bruker" if no role is given.

### PR DESCRIPTION
**Pull request by HenrikEngen**
### Dependencies:
There are no dependencies for this pull request.

### Pictures:
image

### CODE REVIEW

**Comment by petterroea** [comment]
Sorry for feature creeping, but maybe this is a good time to also make the default empty page explain that they are in the wrong place and that they should go to participate instead? It could grow into a dashboard later on(but not what we should use our time on?)

File: src/container/sidebar.jsx [hunk 1]
```diff
@@ -456,8 +456,14 @@ export const Sidebar = () => {
             return named[0].position.name
         }
         else if(mappings.length == 0) {
-            return mappings[0].position.name
-        } else {
+            if(mappings[0] == undefined) {
```
Comment (petterroea) [inline src/container/sidebar.jsx:459]: better to do `mappings.length == 0`

**Comment by petterroea** [comment]
Yes, that's exactly what i suggested.

But no need to spend 10 months making a summary page for logged in users now. Just make it say "hello!" if you have permissions and "wrong place" if not.

File: src/container/sidebar.jsx (hunk 1 above)
Comment (HenrikEngen) [reply src/container/sidebar.jsx]: Done, do I need the else {} code below line 459?

```jsx
454        const named = mappings.filter(mapping => !!mapping.position.name);
455        if(named.length > 0) {
456            return named[0].position.name
457        }
458        else if(mappings.length == 0) {
459            if(mappings.length == 0) {
460                return "Bruker";
461            } else {
462                return mappings[0].position.name
463            }  
464        } 
465        
466        else {
467            return `${mappings.length} stillinger`
468        }
```

File: src/container/sidebar.jsx [hunk 2]
```diff
@@ -456,8 +456,14 @@ export const Sidebar = () => {
             return named[0].position.name
         }
         else if(mappings.length == 0) {
-            return mappings[0].position.name
-        } else {
+            if(mappings.length == 0) {
```
Comment (petterroea) [inline src/container/sidebar.jsx:459]: Why two identical ifs inside each other? Just return "bruker"

**Review by petterroea — CHANGES_REQUESTED** [review]

**Comment by HenrikEngen** [comment]
If the user has no positions:
image

If the user has one or more positions and none of them is a chief position:
image

If the user has one or more positions and one of them is a chief position:
image

If the user has one or more positions and one of them has a custom name (Ex. superadmin):
image

Logic updated and found here:
https://github.com/phoenixlan/phoenixcrew-v1/pull/39/files#diff-06367c7787c9b1e7a7d0c124ec4a2faaba5463c17bd025fd5f79a7b0b4833baeR456-R486

**Review by petterroea — APPROVED** [review]
cool

**Comment by petterroea** [comment]
@HenrikEngen approved, but one change pls: not "crew medlem". We dont have orddeling in Norwegian. Its "crewmedlem".

But even better: "gruppemedlem"?

**Review by bernes1 — APPROVED** [review]

**Comment by HenrikEngen** [comment]
> @HenrikEngen approved, but one change pls: not "crew medlem". We dont have orddeling in Norwegian. Its "crewmedlem".
> 
> But even better: "gruppemedlem"?

@petterroea sweet, thank you! 
Changed from "crew medlem" to "gruppemedlem", merging!

**Pull request merged**